internal/kafka/internal/services/kafka_test.go

```diff
@@ -5011,6 +5011,42 @@ func Test_kafkaService_ManagedKafkasRoutesTLSCertificate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "do not perform a database update if certificate references have not changed",
+			fields: fields{
+				kafkaConfig: &config.KafkaConfig{
+					KafkaDomainName: "some-kafka-domain.bf2.dev",
+				},
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				kafkaTLSCertificateManagementService: &kafkatlscertmgmt.KafkaTLSCertificateManagementServiceMock{
+					IsAutomaticCertificateManagementEnabledFunc: func() bool {
+						return true
+					},
+					ManageCertificateFunc: func(ctx context.Context, domain string) (kafkatlscertmgmt.CertificateManagementOutput, error) {
+						return kafkatlscertmgmt.CertificateManagementOutput{
+							TLSCertRef: "some-cert-ref",
+							TLSKeyRef:  "some-key-ref",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				kafkaRequest: &dbapi.KafkaRequest{
+					Meta: api.Meta{
+						ID: "456",
+					},
+					KafkasRoutesBaseDomainName:      "some-kafka-domain.bf2.dev",
+					KafkasRoutesBaseDomainTLSKeyRef: "some-key-ref",
+					KafkasRoutesBaseDomainTLSCrtRef: "some-cert-ref",
+					Status:                          "ready",
+				},
+			},
+			setupFn: func() {
+				mocket.Catcher.Reset().NewMock().WithQuery(`UPDATE "kafka_requests"`).WithExecException() // should never be called
```

Comment:
By reading these catchers, I believe we want an exception whatever the query is. If that is correct, we can remove this.

Reply:
good catch!